Update pydantic

henrikstranneheim opened the pull request:
## Description

Update to Pydantic versio 2.

### Changed

- Use field_validator and model_validator

### Fixed

- Refactor signature of field_validators

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b update_pydantic -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] See below

### Expected test outcome
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] tests executed by HS
- [x] "Merge and deploy" approved by @seallard 
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

henrikstranneheim commented:
@fevac @ChrOertlin @seallard @ivadym @northwestwitch This one might be of some interest if your planning to update Pydantic in the near future .

ivadym reviewed (approved):
Nice 💯 💯 💯 

Love the new `root_validator`!

seallard reviewed (approved):
Looks good 👍

seallard commented on trailblazer/apps/slurm/models.py:
```diff
@@ -58,19 +59,16 @@ class SqueueResult(BaseModel):
-    def set_jobs_status_distribution(
-        cls, _: Optional[List[SqueueJob]], values: Dict[str, Any]
-    ) -> Optional[Dict[str, float]]:
+    @model_validator(mode="after")
```
Tricky to know which one should be used (before, after or wrap). To me it seems reasonable to always use after? 
https://docs.pydantic.dev/latest/usage/validators/#before-after-wrap-and-plain-validators

diitaz93 reviewed (approved):
Looks good 👍

seallard commented on trailblazer/apps/slurm/models.py:
```diff
@@ -58,19 +59,16 @@ class SqueueResult(BaseModel):
-        for job_status in set(job_statuses):
-            status_distribution[job_status] = round(job_statuses.count(job_status) / len(jobs), 2)
-        return status_distribution
+        job_statuses: List[str] = [job.status for job in self.jobs]
```
We used the values parameter in the old method, I don't see why this would not work but we could also use something like in this [post](https://stackoverflow.com/questions/76734333/pydantic-v2-field-validator-values-argument-equivalent.

The only advantage I can think of is that the `FieldValidationInfo.data` property only contains previously validated fields. So if the `jobs` field had not been validated, it would not be accessible. Not sure how valuable that is. I think I prefer this approach

henrikstranneheim replied:
It will be initialized and validated before being used in the model_validator so it should be safe. 👍

henrikstranneheim commented:
Update to latest Pydantic ✅ 
```
θ67° [henrik.stranneheim@hasta:/home/proj/stage/bin/git/trailblazer] [S_trailblazer] update_pydantic 130 ± pip install --editable .
Obtaining file:///home/proj/stage/bin/git/trailblazer
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Requirement already satisfied: Alchy in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.2.2)
Requirement already satisfied: Click<7 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (6.7)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.3.24)
Requirement already satisfied: alembic in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.11.1)
Requirement already satisfied: coloredlogs in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (15.0.1)
Requirement already satisfied: marshmallow in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (3.19.0)
Requirement already satisfied: requests in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.31.0)
Requirement already satisfied: ruamel.yaml in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.17.31)
Requirement already satisfied: tabulate in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.9.0)
Requirement already satisfied: Flask-Alchy in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.5.0)
Requirement already satisfied: Flask-CORS in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (3.0.10)
Requirement already satisfied: Flask-Reverse-Proxy in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.2.0.2)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.1)
Requirement already satisfied: Flask>=1.1.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.1.4)
Requirement already satisfied: google-auth in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.19.1)
Requirement already satisfied: gunicorn in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (20.1.0)
Requirement already satisfied: markupsafe==2.0.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.0.1)
Requirement already satisfied: cryptography in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (41.0.1)
Requirement already satisfied: pydantic in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.10.9)
Requirement already satisfied: pymysql in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.0.3)
Requirement already satisfied: Werkzeug<2.0,>=0.15 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Flask>=1.1.1->trailblazer==15.2.1) (1.0.1)
Requirement already satisfied: Jinja2<3.0,>=2.10.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Flask>=1.1.1->trailblazer==15.2.1) (2.11.3)
Requirement already satisfied: itsdangerous<2.0,>=0.24 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Flask>=1.1.1->trailblazer==15.2.1) (1.1.0)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Alchy->trailblazer==15.2.1) (7.0.4)
Requirement already satisfied: Mako in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (1.2.4)
Requirement already satisfied: typing-extensions>=4 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (4.6.3)
Requirement already satisfied: importlib-metadata in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (6.6.0)
Requirement already satisfied: importlib-resources in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (5.12.0)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from coloredlogs->trailblazer==15.2.1) (10.0)
Requirement already satisfied: cffi>=1.12 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from cryptography->trailblazer==15.2.1) (1.15.1)
Requirement already satisfied: Six in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Flask-CORS->trailblazer==15.2.1) (1.16.0)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (5.3.1)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (0.3.0)
Requirement already satisfied: rsa<5,>=3.1.4 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (4.9)
Requirement already satisfied: urllib3<2.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (1.26.16)
Requirement already satisfied: setuptools>=3.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from gunicorn->trailblazer==15.2.1) (67.7.2)
Requirement already satisfied: packaging>=17.0 in /home/henrik.stranneheim/.local/lib/python3.8/site-packages (from marshmallow->trailblazer==15.2.1) (20.7)
Requirement already satisfied: charset-normalizer<4,>=2 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from requests->trailblazer==15.2.1) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from requests->trailblazer==15.2.1) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from requests->trailblazer==15.2.1) (2023.5.7)
Requirement already satisfied: ruamel.yaml.clib>=0.2.7 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from ruamel.yaml->trailblazer==15.2.1) (0.2.7)
Requirement already satisfied: pycparser in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from cffi>=1.12->cryptography->trailblazer==15.2.1) (2.21)
Requirement already satisfied: pyparsing>=2.0.2 in /home/henrik.stranneheim/.local/lib/python3.8/site-packages (from packaging>=17.0->marshmallow->trailblazer==15.2.1) (2.4.7)
Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from pyasn1-modules>=0.2.1->google-auth->trailblazer==15.2.1) (0.5.0)
Requirement already satisfied: zipp>=0.5 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from importlib-metadata->alembic->trailblazer==15.2.1) (3.15.0)
Building wheels for collected packages: trailblazer
  Building editable for trailblazer (pyproject.toml) ... done
  Created wheel for trailblazer: filename=trailblazer-15.2.1-0.editable-py3-none-any.whl size=5689 sha256=af9a5bed4919913c3aab3a4306c9ebe6465f09372797f88a6737b0b84a5ce3f9
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-ga9i857l/wheels/6f/bb/55/f6f0099a6c16a60ab41f37fb2e431a0849045dc035b5c498b7
Successfully built trailblazer
Installing collected packages: trailblazer
  Attempting uninstall: trailblazer
    Found existing installation: trailblazer 15.2.1
    Uninstalling trailblazer-15.2.1:
      Successfully uninstalled trailblazer-15.2.1
Successfully installed trailblazer-15.2.1
[henrik.stranneheim@hasta:/home/proj/stage/bin/git/trailblazer] [S_trailblazer] update_pydantic 14s ± pip install --force-reinstall --editable .
Obtaining file:///home/proj/stage/bin/git/trailblazer
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Collecting Alchy (from trailblazer==15.2.1)
  Using cached alchy-2.2.2-py2.py3-none-any.whl (36 kB)
Collecting Click<7 (from trailblazer==15.2.1)
  Using cached click-6.7-py2.py3-none-any.whl (71 kB)
Collecting SQLAlchemy<1.4 (from trailblazer==15.2.1)
  Downloading SQLAlchemy-1.3.24-cp38-cp38-manylinux2010_x86_64.whl (1.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 45.0 MB/s eta 0:00:00
Collecting alembic (from trailblazer==15.2.1)
  Downloading alembic-1.11.2-py3-none-any.whl (225 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.3/225.3 kB 8.7 MB/s eta 0:00:00
Collecting coloredlogs (from trailblazer==15.2.1)
  Using cached coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
Collecting marshmallow (from trailblazer==15.2.1)
  Downloading marshmallow-3.20.1-py3-none-any.whl (49 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.4/49.4 kB 9.7 MB/s eta 0:00:00
Collecting requests (from trailblazer==15.2.1)
  Downloading requests-2.31.0-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.6/62.6 kB 14.5 MB/s eta 0:00:00
Collecting ruamel.yaml (from trailblazer==15.2.1)
  Downloading ruamel.yaml-0.17.32-py3-none-any.whl (112 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112.2/112.2 kB 20.5 MB/s eta 0:00:00
Collecting tabulate (from trailblazer==15.2.1)
  Using cached tabulate-0.9.0-py3-none-any.whl (35 kB)
Collecting Flask-Alchy (from trailblazer==15.2.1)
  Using cached Flask_Alchy-0.5.0-py2.py3-none-any.whl (3.1 kB)
Collecting Flask-CORS (from trailblazer==15.2.1)
  Downloading Flask_Cors-4.0.0-py2.py3-none-any.whl (14 kB)
Collecting Flask-Reverse-Proxy (from trailblazer==15.2.1)
  Using cached flask-reverse-proxy-0.2.0.2.tar.gz (1.2 kB)
  Preparing metadata (setup.py) ... done
Collecting Flask-SQLAlchemy==2.1 (from trailblazer==15.2.1)
  Using cached Flask-SQLAlchemy-2.1.tar.gz (95 kB)
  Preparing metadata (setup.py) ... done
Collecting Flask>=1.1.1 (from trailblazer==15.2.1)
  Downloading Flask-2.3.2-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.9/96.9 kB 12.9 MB/s eta 0:00:00
Collecting google-auth (from trailblazer==15.2.1)
  Downloading google_auth-2.22.0-py2.py3-none-any.whl (181 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 181.8/181.8 kB 17.3 MB/s eta 0:00:00
Collecting gunicorn (from trailblazer==15.2.1)
  Downloading gunicorn-21.2.0-py3-none-any.whl (80 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.2/80.2 kB 11.6 MB/s eta 0:00:00
Collecting markupsafe==2.0.1 (from trailblazer==15.2.1)
  Downloading MarkupSafe-2.0.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (30 kB)
Collecting cryptography (from trailblazer==15.2.1)
  Downloading cryptography-41.0.3-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 92.6 MB/s eta 0:00:00
Collecting pydantic (from trailblazer==15.2.1)
  Downloading pydantic-2.1.1-py3-none-any.whl (370 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 370.9/370.9 kB 33.7 MB/s eta 0:00:00
Collecting pymysql (from trailblazer==15.2.1)
  Downloading PyMySQL-1.1.0-py3-none-any.whl (44 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.8/44.8 kB 4.9 MB/s eta 0:00:00
Collecting Werkzeug>=2.3.3 (from Flask>=1.1.1->trailblazer==15.2.1)
  Downloading werkzeug-2.3.7-py3-none-any.whl (242 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 242.2/242.2 kB 28.5 MB/s eta 0:00:00
Collecting Jinja2>=3.1.2 (from Flask>=1.1.1->trailblazer==15.2.1)
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting itsdangerous>=2.1.2 (from Flask>=1.1.1->trailblazer==15.2.1)
  Using cached itsdangerous-2.1.2-py3-none-any.whl (15 kB)
INFO: pip is looking at multiple versions of flask to determine which version is compatible with other requirements. This could take a while.
Collecting Flask>=1.1.1 (from trailblazer==15.2.1)
  Downloading Flask-2.3.1-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.0/97.0 kB 11.0 MB/s eta 0:00:00
  Downloading Flask-2.3.0-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.0/97.0 kB 11.0 MB/s eta 0:00:00
  Downloading Flask-2.2.5-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.8/101.8 kB 12.0 MB/s eta 0:00:00
  Downloading Flask-2.2.4-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.7/101.7 kB 13.6 MB/s eta 0:00:00
  Downloading Flask-2.2.3-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.8/101.8 kB 14.3 MB/s eta 0:00:00
  Using cached Flask-2.2.2-py3-none-any.whl (101 kB)
  Using cached Flask-2.2.1-py3-none-any.whl (101 kB)
INFO: pip is looking at multiple versions of flask to determine which version is compatible with other requirements. This could take a while.
  Using cached Flask-2.2.0-py3-none-any.whl (101 kB)
  Using cached Flask-2.1.3-py3-none-any.whl (95 kB)
  Using cached Flask-2.1.2-py3-none-any.whl (95 kB)
  Using cached Flask-2.1.1-py3-none-any.whl (95 kB)
  Using cached Flask-2.1.0-py3-none-any.whl (95 kB)
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
  Using cached Flask-2.0.3-py3-none-any.whl (95 kB)
  Using cached Flask-2.0.2-py3-none-any.whl (95 kB)
  Using cached Flask-2.0.1-py3-none-any.whl (94 kB)
  Using cached Flask-2.0.0-py3-none-any.whl (93 kB)
  Using cached Flask-1.1.4-py2.py3-none-any.whl (94 kB)
Collecting Werkzeug<2.0,>=0.15 (from Flask>=1.1.1->trailblazer==15.2.1)
  Using cached Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
Collecting Jinja2<3.0,>=2.10.1 (from Flask>=1.1.1->trailblazer==15.2.1)
  Using cached Jinja2-2.11.3-py2.py3-none-any.whl (125 kB)
Collecting itsdangerous<2.0,>=0.24 (from Flask>=1.1.1->trailblazer==15.2.1)
  Using cached itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
Collecting pydash>=3.4.7 (from Alchy->trailblazer==15.2.1)
  Downloading pydash-7.0.6-py3-none-any.whl (110 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 110.1/110.1 kB 14.7 MB/s eta 0:00:00
Collecting Mako (from alembic->trailblazer==15.2.1)
  Using cached Mako-1.2.4-py3-none-any.whl (78 kB)
Collecting typing-extensions>=4 (from alembic->trailblazer==15.2.1)
  Downloading typing_extensions-4.7.1-py3-none-any.whl (33 kB)
Collecting importlib-metadata (from alembic->trailblazer==15.2.1)
  Downloading importlib_metadata-6.8.0-py3-none-any.whl (22 kB)
Collecting importlib-resources (from alembic->trailblazer==15.2.1)
  Downloading importlib_resources-6.0.1-py3-none-any.whl (34 kB)
Collecting humanfriendly>=9.1 (from coloredlogs->trailblazer==15.2.1)
  Using cached humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
Collecting cffi>=1.12 (from cryptography->trailblazer==15.2.1)
  Downloading cffi-1.15.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (442 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 442.7/442.7 kB 34.2 MB/s eta 0:00:00
Collecting cachetools<6.0,>=2.0.0 (from google-auth->trailblazer==15.2.1)
  Downloading cachetools-5.3.1-py3-none-any.whl (9.3 kB)
Collecting pyasn1-modules>=0.2.1 (from google-auth->trailblazer==15.2.1)
  Downloading pyasn1_modules-0.3.0-py2.py3-none-any.whl (181 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 181.3/181.3 kB 18.1 MB/s eta 0:00:00
Collecting rsa<5,>=3.1.4 (from google-auth->trailblazer==15.2.1)
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting six>=1.9.0 (from google-auth->trailblazer==15.2.1)
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting urllib3<2.0 (from google-auth->trailblazer==15.2.1)
  Downloading urllib3-1.26.16-py2.py3-none-any.whl (143 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.1/143.1 kB 17.7 MB/s eta 0:00:00
Collecting packaging (from gunicorn->trailblazer==15.2.1)
  Downloading packaging-23.1-py3-none-any.whl (48 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 48.9/48.9 kB 7.4 MB/s eta 0:00:00
Collecting annotated-types>=0.4.0 (from pydantic->trailblazer==15.2.1)
  Downloading annotated_types-0.5.0-py3-none-any.whl (11 kB)
Collecting pydantic-core==2.4.0 (from pydantic->trailblazer==15.2.1)
  Downloading pydantic_core-2.4.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB 38.9 MB/s eta 0:00:00
Collecting charset-normalizer<4,>=2 (from requests->trailblazer==15.2.1)
  Downloading charset_normalizer-3.2.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (199 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 199.1/199.1 kB 20.3 MB/s eta 0:00:00
Collecting idna<4,>=2.5 (from requests->trailblazer==15.2.1)
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17 (from requests->trailblazer==15.2.1)
  Downloading certifi-2023.7.22-py3-none-any.whl (158 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.3/158.3 kB 23.3 MB/s eta 0:00:00
Collecting ruamel.yaml.clib>=0.2.7 (from ruamel.yaml->trailblazer==15.2.1)
  Downloading ruamel.yaml.clib-0.2.7-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (555 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 555.3/555.3 kB 33.6 MB/s eta 0:00:00
Collecting pycparser (from cffi>=1.12->cryptography->trailblazer==15.2.1)
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting pyasn1<0.6.0,>=0.4.6 (from pyasn1-modules>=0.2.1->google-auth->trailblazer==15.2.1)
  Downloading pyasn1-0.5.0-py2.py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.9/83.9 kB 13.8 MB/s eta 0:00:00
Collecting zipp>=0.5 (from importlib-metadata->alembic->trailblazer==15.2.1)
  Downloading zipp-3.16.2-py3-none-any.whl (7.2 kB)
Building wheels for collected packages: trailblazer, Flask-SQLAlchemy, Flask-Reverse-Proxy
  Building editable for trailblazer (pyproject.toml) ... done
  Created wheel for trailblazer: filename=trailblazer-15.2.1-0.editable-py3-none-any.whl size=5689 sha256=c9bcd6e1a4f8e90ad1cfe721b24417d2df1d4a781a6d535dd94e40ee956e09f7
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-jmvjjm4d/wheels/6f/bb/55/f6f0099a6c16a60ab41f37fb2e431a0849045dc035b5c498b7
  Building wheel for Flask-SQLAlchemy (setup.py) ... done
  Created wheel for Flask-SQLAlchemy: filename=Flask_SQLAlchemy-2.1-py3-none-any.whl size=13489 sha256=e5fc59c957a5021106cf2c9069f4fe1d3404633051a5beb2da0e1db9b21f2e37
  Stored in directory: /home/henrik.stranneheim/.cache/pip/wheels/c8/51/37/d160618d3a231ded52da2ad1680e7398a16a639744799bf852
  Building wheel for Flask-Reverse-Proxy (setup.py) ... done
  Created wheel for Flask-Reverse-Proxy: filename=flask_reverse_proxy-0.2.0.2-py3-none-any.whl size=2034 sha256=c5255feb4263a038c46ea79d70d6be0ebd2c197c9f741a75e1d16a147de4f71f
  Stored in directory: /home/henrik.stranneheim/.cache/pip/wheels/4e/d5/96/6fdaacb6a85f05f85e4ccc59cf8099acbd29c64c9d25913ee1
Successfully built trailblazer Flask-SQLAlchemy Flask-Reverse-Proxy
Installing collected packages: Flask-Reverse-Proxy, Click, zipp, Werkzeug, urllib3, typing-extensions, tabulate, SQLAlchemy, six, ruamel.yaml.clib, pymysql, pycparser, pyasn1, packaging, markupsafe, itsdangerous, idna, humanfriendly, charset-normalizer, certifi, cachetools, ruamel.yaml, rsa, requests, pydash, pydantic-core, pyasn1-modules, marshmallow, Mako, Jinja2, importlib-resources, importlib-metadata, gunicorn, coloredlogs, cffi, annotated-types, pydantic, google-auth, Flask, cryptography, alembic, Alchy, Flask-SQLAlchemy, Flask-CORS, Flask-Alchy, trailblazer
  Attempting uninstall: Flask-Reverse-Proxy
    Found existing installation: flask-reverse-proxy 0.2.0.2
    Uninstalling flask-reverse-proxy-0.2.0.2:
      Successfully uninstalled flask-reverse-proxy-0.2.0.2
  Attempting uninstall: Click
    Found existing installation: click 6.7
    Uninstalling click-6.7:
      Successfully uninstalled click-6.7
  Attempting uninstall: zipp
    Found existing installation: zipp 3.15.0
    Uninstalling zipp-3.15.0:
      Successfully uninstalled zipp-3.15.0
  Attempting uninstall: Werkzeug
    Found existing installation: Werkzeug 1.0.1
    Uninstalling Werkzeug-1.0.1:
      Successfully uninstalled Werkzeug-1.0.1
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.26.16
    Uninstalling urllib3-1.26.16:
      Successfully uninstalled urllib3-1.26.16
  Attempting uninstall: typing-extensions
    Found existing installation: typing_extensions 4.6.3
    Uninstalling typing_extensions-4.6.3:
      Successfully uninstalled typing_extensions-4.6.3
  Attempting uninstall: tabulate
    Found existing installation: tabulate 0.9.0
    Uninstalling tabulate-0.9.0:
      Successfully uninstalled tabulate-0.9.0
  Attempting uninstall: SQLAlchemy
    Found existing installation: SQLAlchemy 1.3.24
    Uninstalling SQLAlchemy-1.3.24:
      Successfully uninstalled SQLAlchemy-1.3.24
  Attempting uninstall: six
    Found existing installation: six 1.16.0
    Uninstalling six-1.16.0:
      Successfully uninstalled six-1.16.0
  Attempting uninstall: ruamel.yaml.clib
    Found existing installation: ruamel.yaml.clib 0.2.7
    Uninstalling ruamel.yaml.clib-0.2.7:
      Successfully uninstalled ruamel.yaml.clib-0.2.7
  Attempting uninstall: pymysql
    Found existing installation: PyMySQL 1.0.3
    Uninstalling PyMySQL-1.0.3:
      Successfully uninstalled PyMySQL-1.0.3
  Attempting uninstall: pycparser
    Found existing installation: pycparser 2.21
    Uninstalling pycparser-2.21:
      Successfully uninstalled pycparser-2.21
  Attempting uninstall: pyasn1
    Found existing installation: pyasn1 0.5.0
    Uninstalling pyasn1-0.5.0:
      Successfully uninstalled pyasn1-0.5.0
  Attempting uninstall: packaging
    Found existing installation: packaging 20.7
    Uninstalling packaging-20.7:
      Successfully uninstalled packaging-20.7
  Attempting uninstall: markupsafe
    Found existing installation: MarkupSafe 2.0.1
    Uninstalling MarkupSafe-2.0.1:
      Successfully uninstalled MarkupSafe-2.0.1
  Attempting uninstall: itsdangerous
    Found existing installation: itsdangerous 1.1.0
    Uninstalling itsdangerous-1.1.0:
      Successfully uninstalled itsdangerous-1.1.0
  Attempting uninstall: idna
    Found existing installation: idna 3.4
    Uninstalling idna-3.4:
      Successfully uninstalled idna-3.4
  Attempting uninstall: humanfriendly
    Found existing installation: humanfriendly 10.0
    Uninstalling humanfriendly-10.0:
      Successfully uninstalled humanfriendly-10.0
  Attempting uninstall: charset-normalizer
    Found existing installation: charset-normalizer 3.1.0
    Uninstalling charset-normalizer-3.1.0:
      Successfully uninstalled charset-normalizer-3.1.0
  Attempting uninstall: certifi
    Found existing installation: certifi 2023.5.7
    Uninstalling certifi-2023.5.7:
      Successfully uninstalled certifi-2023.5.7
  Attempting uninstall: cachetools
    Found existing installation: cachetools 5.3.1
    Uninstalling cachetools-5.3.1:
      Successfully uninstalled cachetools-5.3.1
  Attempting uninstall: ruamel.yaml
    Found existing installation: ruamel.yaml 0.17.31
    Uninstalling ruamel.yaml-0.17.31:
      Successfully uninstalled ruamel.yaml-0.17.31
  Attempting uninstall: rsa
    Found existing installation: rsa 4.9
    Uninstalling rsa-4.9:
      Successfully uninstalled rsa-4.9
  Attempting uninstall: requests
    Found existing installation: requests 2.31.0
    Uninstalling requests-2.31.0:
      Successfully uninstalled requests-2.31.0
  Attempting uninstall: pydash
    Found existing installation: pydash 7.0.4
    Uninstalling pydash-7.0.4:
      Successfully uninstalled pydash-7.0.4
  Attempting uninstall: pyasn1-modules
    Found existing installation: pyasn1-modules 0.3.0
    Uninstalling pyasn1-modules-0.3.0:
      Successfully uninstalled pyasn1-modules-0.3.0
  Attempting uninstall: marshmallow
    Found existing installation: marshmallow 3.19.0
    Uninstalling marshmallow-3.19.0:
      Successfully uninstalled marshmallow-3.19.0
  Attempting uninstall: Mako
    Found existing installation: Mako 1.2.4
    Uninstalling Mako-1.2.4:
      Successfully uninstalled Mako-1.2.4
  Attempting uninstall: Jinja2
    Found existing installation: Jinja2 2.11.3
    Uninstalling Jinja2-2.11.3:
      Successfully uninstalled Jinja2-2.11.3
  Attempting uninstall: importlib-resources
    Found existing installation: importlib-resources 5.12.0
    Uninstalling importlib-resources-5.12.0:
      Successfully uninstalled importlib-resources-5.12.0
  Attempting uninstall: importlib-metadata
    Found existing installation: importlib-metadata 6.6.0
    Uninstalling importlib-metadata-6.6.0:
      Successfully uninstalled importlib-metadata-6.6.0
  Attempting uninstall: gunicorn
    Found existing installation: gunicorn 20.1.0
    Uninstalling gunicorn-20.1.0:
      Successfully uninstalled gunicorn-20.1.0
  Attempting uninstall: coloredlogs
    Found existing installation: coloredlogs 15.0.1
    Uninstalling coloredlogs-15.0.1:
      Successfully uninstalled coloredlogs-15.0.1
  Attempting uninstall: cffi
    Found existing installation: cffi 1.15.1
    Uninstalling cffi-1.15.1:
      Successfully uninstalled cffi-1.15.1
  Attempting uninstall: pydantic
    Found existing installation: pydantic 1.10.9
    Uninstalling pydantic-1.10.9:
      Successfully uninstalled pydantic-1.10.9
  Attempting uninstall: google-auth
    Found existing installation: google-auth 2.19.1
    Uninstalling google-auth-2.19.1:
      Successfully uninstalled google-auth-2.19.1
  Attempting uninstall: Flask
    Found existing installation: Flask 1.1.4
    Uninstalling Flask-1.1.4:
      Successfully uninstalled Flask-1.1.4
  Attempting uninstall: cryptography
    Found existing installation: cryptography 41.0.1
    Uninstalling cryptography-41.0.1:
      Successfully uninstalled cryptography-41.0.1
  Attempting uninstall: alembic
    Found existing installation: alembic 1.11.1
    Uninstalling alembic-1.11.1:
      Successfully uninstalled alembic-1.11.1
  Attempting uninstall: Alchy
    Found existing installation: alchy 2.2.2
    Uninstalling alchy-2.2.2:
      Successfully uninstalled alchy-2.2.2
  Attempting uninstall: Flask-SQLAlchemy
    Found existing installation: Flask-SQLAlchemy 2.1
    Uninstalling Flask-SQLAlchemy-2.1:
      Successfully uninstalled Flask-SQLAlchemy-2.1
  Attempting uninstall: Flask-CORS
    Found existing installation: Flask-Cors 3.0.10
    Uninstalling Flask-Cors-3.0.10:
      Successfully uninstalled Flask-Cors-3.0.10
  Attempting uninstall: Flask-Alchy
    Found existing installation: Flask-Alchy 0.5.0
    Uninstalling Flask-Alchy-0.5.0:
      Successfully uninstalled Flask-Alchy-0.5.0
  Attempting uninstall: trailblazer
    Found existing installation: trailblazer 15.2.1
    Uninstalling trailblazer-15.2.1:
      Successfully uninstalled trailblazer-15.2.1
Successfully installed Alchy-2.2.2 Click-6.7 Flask-1.1.4 Flask-Alchy-0.5.0 Flask-CORS-4.0.0 Flask-Reverse-Proxy-0.2.0.2 Flask-SQLAlchemy-2.1 Jinja2-2.11.3 Mako-1.2.4 SQLAlchemy-1.3.24 Werkzeug-1.0.1 alembic-1.11.2 annotated-types-0.5.0 cachetools-5.3.1 certifi-2023.7.22 cffi-1.15.1 charset-normalizer-3.2.0 coloredlogs-15.0.1 cryptography-41.0.3 google-auth-2.22.0 gunicorn-21.2.0 humanfriendly-10.0 idna-3.4 importlib-metadata-6.8.0 importlib-resources-6.0.1 itsdangerous-1.1.0 markupsafe-2.0.1 marshmallow-3.20.1 packaging-23.1 pyasn1-0.5.0 pyasn1-modules-0.3.0 pycparser-2.21 pydantic-2.1.1 pydantic-core-2.4.0 pydash-7.0.6 pymysql-1.1.0 requests-2.31.0 rsa-4.9 ruamel.yaml-0.17.32 ruamel.yaml.clib-0.2.7 six-1.16.0 tabulate-0.9.0 trailblazer-15.2.1 typing-extensions-4.7.1 urllib3-1.26.16 zipp-3.16.2
θ65° [henrik.stranneheim@hasta:/home/proj/stage/bin/git/trailblazer] [S_trailblazer] update_pydantic 49s ± down
cd
[henrik.stranneheim@hasta:/home/proj/stage/bin/git/trailblazer] [base] update_pydantic 9s ± cd
[henrik.stranneheim@hasta:~] [base] $ us
[henrik.stranneheim@hasta:~] [S_base] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b update_pydantic -a
Collecting git+https://github.com/Clinical-Genomics/trailblazer@update_pydantic
  Running command git clone --quiet https://github.com/Clinical-Genomics/trailblazer /home/henrik.stranneheim/tmp/pip-req-build-l3ahf0ah
  Cloning https://github.com/Clinical-Genomics/trailblazer (to revision update_pydantic) to ./tmp/pip-req-build-l3ahf0ah
  Running command git checkout -b update_pydantic --track origin/update_pydantic
  Resolved https://github.com/Clinical-Genomics/trailblazer to commit 90428927f41e1fcd1f521b051e36f55d5179c165
  Switched to a new branch 'update_pydantic'
  Installing build dependencies: started
  Branch update_pydantic set up to track remote branch update_pydantic from origin.
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: trailblazer
  Building wheel for trailblazer (pyproject.toml): started
  Building wheel for trailblazer (pyproject.toml): finished with status 'done'
  Created wheel for trailblazer: filename=trailblazer-15.2.1-py3-none-any.whl size=36069 sha256=3aaaf42df9e2c0bc4263caee33bf2b6cd35eac0d079ac21d458171c37dc35230
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-y34donaa/wheels/36/d1/7f/1c1724b44d5e13d8e9393c177adab7ce30c88d8811f02edae9
Successfully built trailblazer
Installing collected packages: trailblazer
  Attempting uninstall: trailblazer
    Found existing installation: trailblazer 15.2.1
    Uninstalling trailblazer-15.2.1:
      Successfully uninstalled trailblazer-15.2.1
Successfully installed trailblazer-15.2.1
Collecting git+https://github.com/Clinical-Genomics/trailblazer@update_pydantic
  Running command git clone --quiet https://github.com/Clinical-Genomics/trailblazer /home/henrik.stranneheim/tmp/pip-req-build-3vtn82rc
  Cloning https://github.com/Clinical-Genomics/trailblazer (to revision update_pydantic) to ./tmp/pip-req-build-3vtn82rc
  Running command git checkout -b update_pydantic --track origin/update_pydantic
  Resolved https://github.com/Clinical-Genomics/trailblazer to commit 90428927f41e1fcd1f521b051e36f55d5179c165
  Switched to a new branch 'update_pydantic'
  Installing build dependencies: started
  Branch update_pydantic set up to track remote branch update_pydantic from origin.
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: Alchy in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.2.2)
Requirement already satisfied: Click<7 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (6.7)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.3.24)
Requirement already satisfied: alembic in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.11.2)
Requirement already satisfied: coloredlogs in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (15.0.1)
Requirement already satisfied: marshmallow in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (3.20.1)
Requirement already satisfied: requests in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.31.0)
Requirement already satisfied: ruamel.yaml in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.17.32)
Requirement already satisfied: tabulate in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.9.0)
Requirement already satisfied: Flask-Alchy in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.5.0)
Requirement already satisfied: Flask-CORS in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (4.0.0)
Requirement already satisfied: Flask-Reverse-Proxy in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (0.2.0.2)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.1)
Requirement already satisfied: Flask>=1.1.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.1.4)
Requirement already satisfied: google-auth in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.22.0)
Requirement already satisfied: gunicorn in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (21.2.0)
Requirement already satisfied: markupsafe==2.0.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.0.1)
Requirement already satisfied: cryptography in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (41.0.3)
Requirement already satisfied: pydantic in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (2.1.1)
Requirement already satisfied: pymysql in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from trailblazer==15.2.1) (1.1.0)
Requirement already satisfied: Werkzeug<2.0,>=0.15 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Flask>=1.1.1->trailblazer==15.2.1) (1.0.1)
Requirement already satisfied: Jinja2<3.0,>=2.10.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Flask>=1.1.1->trailblazer==15.2.1) (2.11.3)
Requirement already satisfied: itsdangerous<2.0,>=0.24 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Flask>=1.1.1->trailblazer==15.2.1) (1.1.0)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from Alchy->trailblazer==15.2.1) (7.0.6)
Requirement already satisfied: Mako in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (1.2.4)
Requirement already satisfied: typing-extensions>=4 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (4.7.1)
Requirement already satisfied: importlib-metadata in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (6.8.0)
Requirement already satisfied: importlib-resources in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from alembic->trailblazer==15.2.1) (6.0.1)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from coloredlogs->trailblazer==15.2.1) (10.0)
Requirement already satisfied: cffi>=1.12 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from cryptography->trailblazer==15.2.1) (1.15.1)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (5.3.1)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (0.3.0)
Requirement already satisfied: rsa<5,>=3.1.4 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (4.9)
Requirement already satisfied: six>=1.9.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (1.16.0)
Requirement already satisfied: urllib3<2.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from google-auth->trailblazer==15.2.1) (1.26.16)
Requirement already satisfied: packaging in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from gunicorn->trailblazer==15.2.1) (23.1)
Requirement already satisfied: annotated-types>=0.4.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from pydantic->trailblazer==15.2.1) (0.5.0)
Requirement already satisfied: pydantic-core==2.4.0 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from pydantic->trailblazer==15.2.1) (2.4.0)
Requirement already satisfied: charset-normalizer<4,>=2 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from requests->trailblazer==15.2.1) (3.2.0)
Requirement already satisfied: idna<4,>=2.5 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from requests->trailblazer==15.2.1) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from requests->trailblazer==15.2.1) (2023.7.22)
Requirement already satisfied: ruamel.yaml.clib>=0.2.7 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from ruamel.yaml->trailblazer==15.2.1) (0.2.7)
Requirement already satisfied: pycparser in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from cffi>=1.12->cryptography->trailblazer==15.2.1) (2.21)
Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from pyasn1-modules>=0.2.1->google-auth->trailblazer==15.2.1) (0.5.0)
Requirement already satisfied: zipp>=0.5 in /home/proj/stage/bin/miniconda3/envs/S_trailblazer/lib/python3.8/site-packages (from importlib-metadata->alembic->trailblazer==15.2.1) (3.16.2)
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
Already on 'update_pydantic'
Already up-to-date.
/home/henrik.stranneheim
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/henrik.stranneheim
done.
Log deploy... done.
trailblazer, version 15.2.1
```
### CLI ✅ 
```
θ61° [henrik.stranneheim@hasta:~] [S_base] 1m50s $ trailblazer scan
slurm_load_jobs error: Invalid job id specified
Error updating analysis for: case - HYG7YDSXX : CalledProcessError
Error updating analysis for: case - gentleamoeba : FileNotFoundError
Error updating analysis for: case - masterimpala : FileNotFoundError
Error updating analysis for: case - starakita : FileNotFoundError
Error updating analysis for: case - goodrodent : FileNotFoundError
Error updating analysis for: case - strongbug : FileNotFoundError
Error updating analysis for: case - masterwombat : FileNotFoundError
Error updating analysis for: case - warmcalf : FileNotFoundError
Error updating analysis for: case - jointstag : FileNotFoundError
Error updating analysis for: case - finerobin : FileNotFoundError
Error updating analysis for: case - wantedhen : FileNotFoundError
Error updating analysis for: case - cleanleech : PermissionError
Error updating analysis for: case - socialearwig : FileNotFoundError
Error updating analysis for: case - topsdodo : FileNotFoundError
Error updating analysis for: case - usablefeline : PermissionError
Error updating analysis for: case - 233471 : FileNotFoundError
Error updating analysis for: case - crispshiner : FileNotFoundError
Error updating analysis for: case - supertiger : PermissionError
Error updating analysis for: case - fullfrog : FileNotFoundError
Error updating analysis for: case - toplouse : FileNotFoundError
Error updating analysis for: case - sunnycondor : FileNotFoundError
Error updating analysis for: case - keyeft : FileNotFoundError
Error updating analysis for: case - robustalien : PermissionError
Error updating analysis for: case - busyowl : PermissionError
Error updating analysis for: case - boldhound : PermissionError
Error updating analysis for: case - vastboar : FileNotFoundError
Error updating analysis for: case - topsworm : PermissionError
Error updating analysis for: case - setcolt : PermissionError
Error updating analysis for: case - proudfish : PermissionError
Error updating analysis for: case - funnyaphid : FileNotFoundError
Error updating analysis for: case - onestag : PermissionError
Error updating analysis for: case - newgopher : PermissionError
Error updating analysis for: case - helpedlab : MissingSqueueOutput
Error updating analysis for: case - goodbengal : MissingSqueueOutput
Error updating analysis for: case - grandaphid : PermissionError
All analyses updated!
```

henrikstranneheim commented:
### API ✅ 
<img width="1361" alt="Screenshot 2023-08-14 at 13 09 23" src="https://github.com/Clinical-Genomics/trailblazer/assets/584738/f1002de3-b3e6-464f-b89d-8bd95240c0fa">
